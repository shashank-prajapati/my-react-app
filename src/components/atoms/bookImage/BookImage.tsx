import React from "react";

type Props = {
  source: string;
  width?: string | number | undefined
};

function BookImage(props: Props) {
  const { source,width } = props;
  return (
    <img
      style={{ display: "block" }}
      src={source}
      width={width}
      alt="Book Cover"
    ></img>
  );
}

export default BookImage;
