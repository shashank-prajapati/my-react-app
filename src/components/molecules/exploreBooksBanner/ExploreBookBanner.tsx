import { Grid, Paper, Typography } from '@mui/material';
import React from 'react';
import ReadingImage from '../../atoms/readingImage/ReadingImage';

type Props = {};

const ExploreBookBanner = ({}: Props)=> {
  return <Paper sx={{
      bgcolor:'footerBgColor.main',
      px:9,
      py:4
  }}>
      <Grid container>
        <Grid item sm={6} sx={{py:9}}>
            <Grid container rowSpacing={4}>
                <Grid item sm={12}>
                    <Typography variant='heading' color='navTextColors.main'>Explore Books on entrepreneurship</Typography>
                </Grid>
                <Grid item sm={12}>
                    <Typography variant='subtitle2' color='extraTextColor.main'>Everything you need to know about thriving on a shoestring budget, making your first million, and hiring right from the start.</Typography>
                </Grid>
            </Grid>
        </Grid>
        <Grid item sm={6} sx={{
            display:'flex',
            justifyContent:'center'
        }}>
            <ReadingImage/>
        </Grid>
      </Grid>
  </Paper>;
}

export default ExploreBookBanner;
