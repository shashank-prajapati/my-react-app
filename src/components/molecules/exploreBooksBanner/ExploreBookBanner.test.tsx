import * as React from "react";
import { render, fireEvent ,screen} from "@testing-library/react";
import ExploreBookBanner from "./ExploreBookBanner";
import { Typography } from "@mui/material";
import BookImage from "../../../media/coverImage.png";

describe("Testing Explore Banner", () => {
  it("should match the Explore Banner snapshot", () => {
    const { container } = render(
      <ExploreBookBanner/>
    );
    expect(container).toMatchSnapshot();
  });
});
