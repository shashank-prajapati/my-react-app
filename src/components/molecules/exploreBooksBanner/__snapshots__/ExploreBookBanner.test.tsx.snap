// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Explore Banner should match the Explore Banner snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-ulgjkv-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6 css-17vwnak-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-sv021s-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 css-13shxg4-MuiGrid-root"
          >
            <span
              class="MuiTypography-root MuiTypography-heading css-g6ah8q-MuiTypography-root"
            >
              Explore Books on entrepreneurship
            </span>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 css-13shxg4-MuiGrid-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1gwt0ci-MuiTypography-root"
            >
              Everything you need to know about thriving on a shoestring
budget, making your first million, and hiring right from the
start.
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6 css-fxm7ij-MuiGrid-root"
      >
        <img
          alt="Book"
          height="inherit"
          src="readingImage.png"
          style="display: block;"
          width="inherit"
        />
      </div>
    </div>
  </div>
</div>
`;
